src/ibex_bluesky_core/callbacks/__init__.py

```diff
@@ -1 +1,203 @@
+            add_table_cb: whether to add a table callback.
+            add_peak_stats: whether to add a peak stats callback.
+            show_fit_on_plot: whether to show fit on plot.
+            ax: An optional axes object to use for plotting.
+
+        """  # noqa
+        if measured_fields is None:
+            measured_fields = []
+        if fields_for_live_table is None:
+            fields_for_live_table = []
+        if fields_for_hr_file is None:
+            fields_for_hr_file = []
+
+        if show_fit_on_plot and (not add_fit_cb or fit is None):
+            raise ValueError(
+                "Fit has been requested to show on plot without a fitting method or callback."
+            )
+        if (add_peak_stats or add_fit_cb or add_plot_cb) and (x is None or y is None):
+            raise ValueError(
+                "X and/or Y not specified when trying to add a plot, fit or peak stats."
+            )
+
+        self.subs = []
+        if add_human_readable_file_cb:
+            combined_hr_fields = measured_fields + fields_for_hr_file
+            if not combined_hr_fields:
+                raise ValueError("No fields specified for the human-readable file")
+            self.subs.append(
+                HumanReadableFileCallback(
+                    fields=combined_hr_fields,
+                    output_dir=Path(human_readable_file_output_dir)
+                    if human_readable_file_output_dir
+                    else DEFAULT_PATH,
+                ),
+            )
+
+        if add_table_cb:
+            combined_lt_fields = measured_fields + fields_for_live_table
+            if not combined_lt_fields:
+                raise ValueError("No fields specified for the live table")
+            self.subs.append(
+                LiveTable(combined_lt_fields),
+            )
+
+        if add_peak_stats:
+            self._peak_stats = PeakStats(x=x, y=y)
+            self.subs.append(self._peak_stats)
+
+        if (add_plot_cb or show_fit_on_plot) and not ax:
+            logger.debug("No axis provided, creating a new one")
+            fig, ax, exc, result = None, None, None, None
+            done_event = threading.Event()
+
+            class _Cb(QtAwareCallback):
+                def start(self, doc: RunStart) -> None:
+                    nonlocal result, exc, fig, ax
+                    try:
+                        plt.close("all")
+                        fig, ax = plt.subplots()
+                    finally:
+                        done_event.set()
+
+            cb = _Cb()
+            cb("start", {"time": 0, "uid": ""})
+            done_event.wait(10.0)
+
+        if add_fit_cb:
+            if fit is None:
+                raise ValueError("fit method must be specified if add_fit_cb is True")
+            self._live_fit = LiveFit(fit, y=y, x=x, yerr=yerr)  # pyright: ignore reportArgumentType
+            if show_fit_on_plot:
+                self.subs.append(LiveFitPlot(livefit=self._live_fit, ax=ax))
```

Comment:
```suggestion
            if show_fit_on_plot:
                self.subs.append(LiveFitPlot(livefit=self._live_fit, ax=ax))
            else:
                self.subs.append(self._live_fit)
```

Otherwise I can't see that fit ever gets subscribed?